extensions/resource-deployment/src/ui/validation/validations.ts

```diff
@@ -0,0 +1,216 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import { EOL } from 'os';
+import { throwUnless } from '../../common/utils';
+import * as loc from '../../localizedConstants';
+import { InputComponent } from '../modelViewUtils';
+import { RadioGroupLoadingComponentBuilder } from '../radioGroupLoadingComponentBuilder';
+
+export interface ValidationResult {
+	valid: boolean;
+	message?: string;
+}
+
+export type Validator = () => Promise<ValidationResult>;
+export type VariableValueGetter = (variable: string) => Promise<string | number | undefined | null>;
```

Comment:
where is null value possibly coming from? can we simplify this to Promise<string | number | undefined>

Reply:
sure.

Reply:
removed null.